### PR TITLE
Disable "Save" and "Save as" when datastore is null

### LINKED ISSUE #1498: Disable "save" actions when no job is opened
On the main screen of DC desktop you can click "Save" or "Save as" even if you're not building a job right now. 

no-job-save

I think these actions should be disabled when there is nothing to save. 

EDIT: They are disabled after you start DC. But when you open a job and close it again, these actions stay enabled and that is the issue here.

### PR DESCRIPTION
**Pull request by ClaudiaPHI**
fixes #1498

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java [hunk 1]
```diff
@@ -908,7 +908,7 @@ public void applyPropertyValues() {
     }
 
     private void onSourceColumnsChanged() {
-        boolean everythingEnabled = !_analysisJobBuilder.getSourceColumns().isEmpty();
+        boolean everythingEnabled = _datastore != null;
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: This seems wrong. Shouldn't it be _both_?

**Comment by LosD** [comment]
Travis error is seemingly unrelated (stalled build)

File: desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java (hunk 1 above)
Comment (ClaudiaPHI) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: There is no reason to be both.  It works perfectly when I tested.

**Comment by jhorcicka** [comment]
The first difference is just after starting a new empty job. In earlier version save buttons were allowed but clicking on them caused an exception ("fix job problems to save it"). Now the buttons are disabled until a datastore is added to the graph.

**Comment by LosD** [comment]
Well spotted! So we should keep both the original check and the null check, then.

File: desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java [hunk 2]
```diff
@@ -908,7 +908,7 @@ public void applyPropertyValues() {
     }
 
     private void onSourceColumnsChanged() {
-        boolean everythingEnabled = !_analysisJobBuilder.getSourceColumns().isEmpty();
+        boolean everythingEnabled = !_analysisJobBuilder.getSourceColumns().isEmpty() &&  _datastore != null;
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: Now I'm totally being a pain in the behind, but there's a double space between `&&` and `_datastore`

Comment (ClaudiaPHI) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: done

**Comment by LosD** [comment]
LGTM! Let's merge when Travis is happy

**Pull request merged**